docs/processes/delivery/Continous/github-action/github-action-for-wordpress.md

```diff
@@ -0,0 +1,238 @@
+
+
+##### 2. Via custom YML code _(within project setup phase)_:
+_You can manually create a respective YML file and put it into the GitHub workflows folder_
+
+In this case, follow the below steps to set up
+1. Make sure you are on the master/main branch of the project _(the file should exist in your master and development branch and it should not be in gitignore)_
+2. Create a folder `.github` inside create `workflows` folder in the root directory of the project _(in our case it will reside outside the wp-content folder)_
+3. Under the workflows folder, you can create your deployment file which is a YML file in the end the path for the file will look like this: `.github/workflows/dev.yml`
+4. You can set up all the processes and actions to be taken once the GitHub action is triggered.
+5. Once you are done with the YML file commit the changes and push the file
+6. As per the code, the actions will be taken
+7. Later step you will get to know the details of what goes into the YML file
+
+<br>
+
+#### <u>Step 3</u>: Understanding the YML script below
+Below attached is the snippet for the basic flow which is set for CI/CD in our case
+<u>**Note:**</u> the code is case sensitive and the indentation is very important wrong indentation will result in **Build Failure**
+
+```yml 
+# This is a basic workflow to help you get started with Actions
+name: Auto Deployment on Dev Environment
+ 
+# Controls when the workflow will run
+on:
+ # Triggers the workflow on push or pull request events but only for the staging branch
+ push:
+   branches: [ staging ]
+  
+jobs:
+ deploy:
+   runs-on: ubuntu-latest
+ 
+   steps:       
+   # Deploy Files on the server using rsync command, after that delete the cache folder from server
+   - uses: actions/checkout@v2
+   - name: Rsync to WP Engine Dev Env
+     env:
+       dest: 'dev123v@dev123v.ssh.wpengine.net'
+     run: |
+       echo "${{secrets.DEV_DEPLOY_KEY_PRIVATE}}" > deploy_key
+       chmod 600 ./deploy_key
+       rsync -chav -e 'ssh -i ./deploy_key -o StrictHostKeyChecking=no' \
+         --exclude /deploy_key \
+         --exclude /.git/ \
+         --exclude /.gitignore/ \
+         --exclude /.github/ \
+         --exclude /.github/workflows \
+         --exclude /node_modules/ \
+         ./ ${{env.dest}}:/sites/dev123v
+       ssh -i ./deploy_key -o StrictHostKeyChecking=no ${{env.dest}} 'rm -r /sites/dev123v/wp-content/cache -f'
+```
+</br>
+
+For more information about the GitHub action please visit this [link](https://github.blog/2021-11-04-10-github-actions-resources-basics-ci-cd/)
+Documentation for GitHub action: [link](https://docs.github.com/en/actions)
+
+
+
+Here is the high-level summary/understanding of the above code (github _[link](https://github.com/axioned/axioned-wordpress-starter/blob/main/.github/workflows/dev.yml))_:
+
+
+- **on**: Since we are using GitHub actions to deploy the files from the GitHub server/repository to the development server _(where the site is hosted)_
```

Comment:
Can we paste the line for code instead of mentioning the key name 'on'?
This can improves the readability of the doc and easier to follow while reading.
e.g. 
```
on:
 push:
```